tests/virt_autotest/sriov_network_card_pci_passthrough.pm

```diff
@@ -0,0 +1,346 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+# 
+# Summary: A test to pass SR-IOV Ethernet VFs to guest via libvirt. Both KVM & XEN hosts are supported.
+# Maintainer: Julie CAO <JCao@suse.com>
+ 
+use base "virt_feature_test_base";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+use utils;
+use virt_utils;
+use xen;
+use virt_autotest::virtual_network_utils qw(save_guest_ip test_network_interface);
+ 
+sub run_test {
+    my $self = shift;
+
+    #set up ssh, packages and iommu on host
+    prepare_host();
+ 
+    #get the SR-IOV device BDF and interface
+    my @host_pfs;
+    @host_pfs = get_sriov_ethernet_devices();
+    record_info("Find SR-IOV devices", "@host_pfs");
+
+    #get/set nessisary variables for test
+    my $gateway = script_output "ip r s | grep 'default via' | cut -d ' ' -f3";
+    set_var("SRIOV_NETWORK_CARD_PCI_PASSSHTROUGH", 1);    #to differenciate virtual network tests
+
+    # enable 8 vfs for the SR-IOV device on host
+    my @host_vfs = enable_vf(@host_pfs);
+    record_info("VFs enabled", "@host_vfs");
+
+    foreach my $guest (keys %xen::guests) {
+
+        prepare_guest($guest);
+    
+        #passthrough 2 vf ethernet devices from host
+        my @vfs               = ();
+        my $passthru_vf_count = 2;           #the number of vf count to be passed through to guests
+        for (my $i = 0; $i < $passthru_vf_count; $i++) {
+
+            my %vf;
+
+            #detach the vf from host
+            $vf{host_bdf} = $host_vfs[int(rand($#host_vfs + 1))];
+            for (my $j = 0; $j < $i; $j++) {
```

Comment:
When i=0 and j=0  j<i ?

Reply:
Yes, to make sure each vf in the list is unique. for the first vf(i=0), no need to check, then skip the for loop.